app/helpers/application_helper/toolbar/basic.rb

```diff
@@ -1,66 +1,34 @@
-class ApplicationHelper::Toolbar::Basic
-  include Singleton
-  extend SingleForwardable
-  delegate %i(api_button button select twostate separator definition button_group custom_content) => :instance
-
-  attr_reader :definition
-
-  def custom_content(name, args)
-    @definition[name] = ApplicationHelper::Toolbar::Custom.new(name, args)
-  end
-
-  def button_group(name, buttons)
-    @definition[name] = ApplicationHelper::Toolbar::Group.new(name, buttons)
-  end
-
-  def api_button(id, icon, title, text, keys = {})
-    keys[:data] = {
-      'function'      => 'sendDataWithRx',
-      'function-data' => {
-        'type'       => "generic",
-        'controller' => "toolbarActions",
-        'payload'    => {
-          'entity' => keys[:api][:entity].pluralize,
-          'action' => keys[:api][:action]
-        }
-      }.to_json
-    }
-    generic_button(:button, id, icon, title, text, keys)
+class ApplicationHelper::Toolbar::Basic < ApplicationHelper::Toolbar::Base
+  def definition(record = nil)
+    extension_classes_filtered(record).reduce(@definition) do |acc, ext|
+      acc.merge(ext.definition)
+    end
   end
 
-  def button(id, icon, title, text, keys = {})
-    generic_button(:button, id, icon, title, text, keys)
-  end
+  private
 
-  def select(id, icon, title, text, keys = {})
-    generic_button(:buttonSelect, id, icon, title, text, keys)
+  def extension_classes
+    @extension_classes ||= load_extension_classes
   end
 
-  def twostate(id, icon, title, text, keys = {})
-    generic_button(:buttonTwoState, id, icon, title, text, keys)
-  end
+  def extension_classes_filtered(record)
+    return extension_classes if record.nil?
 
-  def separator
-    {:separator => true}
+    # Example:
+    # "ManageIQ::Providers::Amazon::ToolbarOverrides::EmsCloudCenter" is a match for
+    #   "ManageIQ::Providers::Amazon::CloudManager
+    extension_classes.find_all do |ext|
+      ext.name.split('::')[0..-3] == record.class.name.split('::')[0..-2]
```

Comment:
@Ladas, @slemrmartin : Do we have a better way to do the matching?

Reply:
You compare just the ```ManageIQ::Providers::Amazon```, right? So maybe 
```ext.name.split('::')[0..2] == record.class.name.split('::')[0..2]``` looks less fragile

Not sure about a better way